src/gretel_trainer/relational/multi_table.py

```diff
@@ -516,6 +510,35 @@ def _create_debug_summary(self) -> None:
             self._project, str(debug_summary_path)
         )
 
+    def delete_models(
+        self,
+        workflow: str,
+        *,
+        only: Optional[list[str]] = None,
+        ignore: Optional[list[str]] = None,
+    ) -> None:
+        only, ignore = self._get_only_and_ignore(only, ignore)
+        tables = [
+            table
+            for table in self.relational_data.list_all_tables()
+            if not skip_table(table, only, ignore)
+        ]
+        if workflow == "classify":
```

Comment:
This is the first time users pass a workflow "name" as an argument—until now it's all been method names (`classify`, `train_transforms`, etc.) or attributes (`synthetic_output_tables`, `transform_output_tables`). I'm almost tempted to continue that pattern here and change this from `delete_models(workflow: str, ...)` to three methods
```
delete_classify_models
delete_transform_models
delete_synthetics_models
```
Thoughts?

Reply:
Could this `delete_models` clean all of the state? Are there use cases where you'd want to clean up one, but not all of the workflows?

---

And a related question - I was trying to find a test that would train synthetics and transforms in the same `MultiTable` instance, but didn't find one. Is use case like that supported/makes sense?

Reply:
> Are there use cases where you'd want to clean up one, but not all of the workflows?

In fairness, I'm not really aware of any specific use case for deleting either some or all models or workflows. I added this because it felt like an appropriate mechanism to have based on the (new/emerging) behavior of users being able to add tables to their set of trained tables, plus it being pretty easy to do.

Earlier, if you called `train` multiple times you would 1) train models for all tables both times, and 2) orphan the first set of models (they'd still exist in the project, but MultiTable would replace internal pointers to them with pointers to the second set of trained models).
```python
mt.train()
# mt._synthetics_train.models == {"users": "model_id_123", "events": "model_id_abc"}
mt.train()
# mt._synthetics_train.models == {"users": "model_id_789", "events": "model_id_xyz"}

# 4 models exist in the project
```

Now, depending on only/ignore args, you _could_ do something like this:
```python
mt.train_synthetics(only=["users"])
# mt._synthetics_train.models == {"users": "model_id_123"}
mt.train_synthetics(only=["events"])
# we didn't abandon the users model...
# mt._synthetics_train.models == {"users": "model_id_123", "events": "model_id_abc"}


# ...but if we call train again with users in scope, we'd abandon it like before
mt.train_synthetics(only=["users"])
# mt._synthetics_train.models == {"users": "model_id_789", "events": "model_id_abc"}
```
Something about this makes me feel like having a mechanism for deleting models should exist from a "library completeness" perspective. However, I'm open to the argument that this isn't a real requirement / the use case is too unclear at this point and that we should just drop this new `delete_models` method altogether / save it for later.

Reply:
Update: I decided to remove this entirely for now, we can revisit if/when a well-defined use case for it arises.